CMakeLists.txt

```
@@ -17,12 +16,73 @@ endif()
 find_package(ament_cmake REQUIRED)
 find_package(rosidl_default_generators REQUIRED)
 
+set(protobuf_MODULE_COMPATIBLE TRUE)
+find_package(Protobuf CONFIG REQUIRED)
+message(STATUS "Using protobuf ${Protobuf_VERSION}")
+
+find_package(gRPC CONFIG REQUIRED)
+message(STATUS "Using gRPC ${gRPC_VERSION}")
+
+file(MAKE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/proto)
+add_custom_command(
+  OUTPUT
+    "${CMAKE_CURRENT_BINARY_DIR}/proto/ninshiki.pb.cc"
+    "${CMAKE_CURRENT_BINARY_DIR}/proto/ninshiki.pb.h"
+    "${CMAKE_CURRENT_BINARY_DIR}/proto/ninshiki.grpc.pb.cc"
+    "${CMAKE_CURRENT_BINARY_DIR}/proto/ninshiki.grpc.pb.h"
+  COMMAND $<TARGET_FILE:protobuf::protoc>
+  ARGS
+    --grpc_out "${CMAKE_CURRENT_BINARY_DIR}/proto"
+    --cpp_out "${CMAKE_CURRENT_BINARY_DIR}/proto"
+    -I ${CMAKE_CURRENT_SOURCE_DIR}/proto
+    --plugin=protoc-gen-grpc=$<TARGET_FILE:gRPC::grpc_cpp_plugin>
+    ${CMAKE_CURRENT_SOURCE_DIR}/proto/ninshiki.proto
+  DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/proto/ninshiki.proto
+)
+
+add_library(ninshiki_proto SHARED
```

I suggest not making the `ninshiki_proto` a shared library, if possible:

```suggestion
add_library(ninshiki_proto
```